laser_encoder.py

```diff
@@ -0,0 +1,93 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Optional, List, Any, Iterable
+import numpy as np
+import torch
+from jina import Executor, DocumentArray, requests
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:i + batch_size]
+
+
+class LaserEncoder(Executor):
+    """
+    Encode an array of string in size `B` into an ndarray in size `B x D`
+
+    The ndarray potentially is BatchSize x (Channel x Height x Width)
+
+    :class:`LaserEncoder` is a encoder based on Facebook Research's LASER
+    (Language-Agnostic SEntence Representations) to compute multilingual
+    sentence embeddings: https://github.com/facebookresearch/LASER
+    :param path_to_bpe_codes: path to bpe codes from Laser.
+        Defaults to Laser.DEFAULT_BPE_CODES_FILE.
+    :param path_to_bpe_vocab: path to bpe vocabs from Laser.
+        Defaults to Laser.DEFAULT_BPE_VOCAB_FILE.
+    :param path_to_encoder: path to the encoder from Laser.
+        Defaults to Laser.DEFAULT_ENCODER_FILE.
+    :param default_batch_size: size of each batch
+    :param default_traversal_paths: traversal path of the Documents, (e.g. 'r', 'c')
+    :param on_gpu: set to True if using GPU
+    :param language: language of the text. Defaults to english(en).
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+            self,
+            path_to_bpe_codes: Optional[str] = None,
+            path_to_bpe_vocab: Optional[str] = None,
+            path_to_encoder: Optional[str] = None,
+            on_gpu: bool = False,
+            default_batch_size: int = 32,
+            default_traversal_paths: List[str] = 'r',
+            language: str = 'en',
+            *args,
+            **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        from laserembeddings import Laser
+        self._path_to_bpe_codes = path_to_bpe_codes or Laser.DEFAULT_BPE_CODES_FILE
+        self._path_to_bpe_vocab = path_to_bpe_vocab or Laser.DEFAULT_BPE_VOCAB_FILE
+        self._path_to_encoder = path_to_encoder or Laser.DEFAULT_ENCODER_FILE
+        self.on_gpu = on_gpu
+        self.default_batch_size = default_batch_size
+        self.default_traversal_paths = default_traversal_paths
+        self.language = language.lower()
+
+        from laserembeddings import Laser
```

Comment:
why twice? Lets import it on the top, no need to hide here anymore

Reply:
done